docs/07_how-to/05_ingest-csv.mdx

```diff
@@ -0,0 +1,288 @@
+    camelSource { 
+        location = file://run/runtime/coppClarkSimple
+    }
+)
+```
+
+### Operators
+Operators are used to transform input format to output format (although they can do more). You can chain them together, where each operator simply takes the output type of the previous one. 
+
+So in this case, the output of `.split` goes into `.map` and the output type of `.map` is a list of DbOperations, which the next sink (dbSync) can take as input.
+
+```kotlin
+            //1
+            .split(csvRawDecoder())
+            //2
+            .map { row ->
+                DbOperation.Upsert( ExchangeHoliday {
+                    //The exclamations at the end tell the code "we expect this to always be present in the row data"
+                    holidayDate = parseCCDateFormat(row.data["EventDate"])!!
+                    holidayName = row.data["EventName"]
+                    holidayType = getHolidayType(row.data["FileType"])!!
+                    isoMic = row.data["ISO MIC Code"]!!
+                })
+            }
+```
+
+The example above has two operators. Let's look at them more closely.
+
+- `split` uses the pre-built decoder which converts the CSV file and outputs a map containing each row in column + value: `CsvRow<Map<String, String>>`
+- `map` takes the data we need from the row columns and formats it accordingly. We are building up ExchangeHoliday, a database entity in our app, wrapped in the Db operation we want to perform on it; in this case, we want to upsert.
+
+For more complex requirements, you can create your own [transformer](#keeping-the-data-transformation-separate) in a seperate file. You can see this in two of our examples.
+
+Also, you can build multiple database entities from a single file. You can see details of this in the [third example](#updating-multiple-different-entities-from-a-single-source).
+
+### Sink
+Once the data is in the required format, we want to sink it. For file ingress, this means updating the database. 
+
+The example sink below takes the output from `.map` (a set of DB operations) and handles the updating of the database.
+
+```kotlin
+.sink(dbSink())
+```
+
+That’s all you need to ingest the data and put it in your database.
+
+However, there are additional helpers for things such as `onCompletion`. This enables you specify what happens after the file processing is complete. 
+
+In the example below, we use this to log that we’ve completed it, and to send a notification to users:
+
+```kotlin
+.onCompletion { context ->
+    LOG.info("Completed processing of file ${context.data.name}")
+}.onCompletion(
+    notifyAllScreensOnCompletion {
+        body = "Finished processing filename ${context.data.name}"
+        header = "Finished Processing Copp Clark Exchange Holidays File"
+        severity = Severity.Information
+    }
+)
+```
+
+[comment]: <> (Waiting on error handling to add a section on it here)
+
+### What can you ingest?
+You can take data from the following sources:
```

Comment:
Not sure we need this section. 
1. Not sure it is accurate
2. Page is about csv ingress